### DISCUSSION TITLE
On device AI support for Android and Apple

### DISCUSSION THREAD

**Discussion by sej69** [CommunityToolkit/Maui#2021] [New Feature Discussions]
Both Apple (Fall of 2024) and Android (May 14th, 2024) are deploying AI services on their devices.  These services would allow developers to make their applications more intelligent.  AI is where everything is going, getting a head start on being able to develop apps using these services will make .NET Maui more relevant in the future.

**Comment by bijington** [comment]
Do you have any reference to the documentation and features that the platforms provide? Or some form of proposal for what should be exposed in terms of a common API?

**Reply by sej69** [reply]
Android's is here: https://developer.android.com/ai/generativeai#:~:text=Google%20offers%20multiple%20generative%20AI,use%20in%20your%20Android%20app.&text=For%20complete%20details%20on%20each,rate%20limits%2C%20see%20Gemini%20models.

Apple is here:
https://developer.apple.com/machine-learning/
